### PR TITLE
Refactor the area loading internal function

### PR DESCRIPTION
**Pull request by mraspaud**
~This PR adds the possibility to create an area definition from a yaml string. So `create_area_def_from_dict` and `create_area_def_from_yaml` utility functions were added.~

This PR refactors the area loading a little.

### CODE REVIEW

**Comment by gerritholl** [comment]
How is this different from `load_area_from_string`?

**Comment by mraspaud** [comment]
it probably isn't, I wasn't aware it existed :) I'll have a look

**Comment by mraspaud** [comment]
Sorry about that @gerritholl , I was so sure this didn't exist already. Anyway, I removed the meat of this and just left the small refactoring.

File: pyresample/test/test_area_config.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
```
Comment (gerritholl) [inline pyresample/test/test_area_config.py:3]: Do these lines actually do anything useful for a test module?  As far as I know, the first line only matters for commandline scripts and the second is redundant in Python 3?

**Review by gerritholl — APPROVED** [review]
Looks good to me, just one minor question.

File: pyresample/test/test_area_config.py (hunk 1 above)
Comment (mraspaud) [reply pyresample/test/test_area_config.py:3]: I haven't thought about that tbh. For consistency, maybe we should have a PR just for removing all of these?

Comment (mraspaud) [reply pyresample/test/test_area_config.py:3]: Or I could do it here...

Comment (djhoese) [reply pyresample/test/test_area_config.py:3]: We could probably remove them. I've only ever added them for consistency. As a library, we could probably remove the usage of both in all pytroll packages.

**Review by djhoese — APPROVED** [review]
LGTM

**Comment by djhoese** [comment]
Regarding the test failures:

This has to do with new versions of setuptools. I would think we should be fine, but obviously things aren't working. It looks like there has been a patch release of setuptools since the last CI run here. It may be worth restarting them @mraspaud but I wanted to see what you thought. I can't tell from the output what version of setuptools was used in the test install. Or I wonder if this requires a new version of versioneer.

**Comment by djhoese** [comment]
Ah, there was a versioneer release to fix it: https://github.com/python-versioneer/python-versioneer/releases/tag/0.23

**Comment by mraspaud** [comment]
Updated versioneer, let's see how it goes

**Pull request merged**